Rakefile

```diff
@@ -262,6 +263,399 @@ task :xcode => [:dependencies] do
+          #if xcode is not installed, prompt user to install and terminate rake
+          puts "XCode not Found!"
+          puts ""
+          puts "====================================================================================="
+          puts "Developing for WordPressiOS requires XCode."
+          puts "Please install XCode before setting up WordPressiOS"
+          puts "https://apps.apple.com/app/xcode/id497799835?mt=12"
+          abort("")
+        else
+          puts "Xcode installed"
+        end
+
+        puts "Checking CI recommendded installed XCode version"
+
+        if !xcode_version_is_correct?
+          #if xcode is the wrong version, prompt user to install the correct version and terminate rake
+          puts "Not recommended version of XCode installed"
+          puts "It is recommended to use Xcode version #{get_ci_xcode_version}"
+          STDOUT.puts "Please press enter to continue"
+          complete = STDIN.gets.strip
+          next
+        end
+      end
+
+      #Check if XCode is installed
+      def xcode_installed?
+        system "xcodebuild -version", [:out, :err] => File::NULL
+      end
+
+      #compare xcode version to expected CI spec version
+      def xcode_version_is_correct?
+        if get_xcode_version == get_ci_xcode_version
+          puts "Correct version of XCode installed"
+          return true
+        end
+      end
+
+      #get xcode version
+      def get_xcode_version
+        puts 'Checking installed version of XCode'
+        version = %x[xcodebuild -version]
+
+        version.split(" ")[1]
+      end
+
+      def get_ci_xcode_version
+        ci_config = File.read(".circleci/config.yml")
+        specs = YAML.load(ci_config)
+
+        ci_version = specs["jobs"]["Build Tests"]["executor"]["xcode-version"]
+      end
+    #End namespace xcode-app
+    end
+
+    #XCode-select command line tools must be installed to update dependencies
+    #xcode_select checks the existence of xcode-select on developer's machine, installs if not found
+    namespace :xcode_select do
+      task :check do
+        puts "Checking system for XCode-select"
+        unless command?("xcode-select")
+          Rake::Task["xcode:xcode_select:install"].invoke
+        else
+          puts "XCode-select installed"
+        end
+      end
+
+      task :install do
+        puts "Installing xcode select"
+        sh "xcode-select --install"
+      end
+    #End namesapce :xcode-select
+    end
+  #End namespace xcode
+  end
+
+  #Tools namespace deals with installing developer and OSS tools required to work on WPiOS
+  namespace :tools do
+    task :check_oss => %w[homebrew:check addons:check_oss]
+    task :check_developer => %w[homebrew:check addons:check_developer]
+
+    #Check for Homebrew and install if missing
+    namespace :homebrew do
+      task :check do
+        puts "Checking system for Homebrew"
+        unless command?("brew")
+          Rake::Task["tools:homebrew:install"].invoke
```

Comment:
How would you feel about asking the user permission to run this?

Reply:
Great idea!  Added